include/real/exact_number.hpp

```diff
@@ -693,190 +694,322 @@ namespace boost {
+                    denominator = denominator * _2;
+                    numerator = numerator * _2;
+                }
+                /* preprocessing end */
+
+                /* Reciprocal computation starts */
+                /* initializing the reciprocal guess */
+                /* initial guess for the reciprocal is (48 - 32*divisor)/17 */
+                exact_number<T> reciprocal, more_precise_reciprocal;
+                exact_number<T> _32, _48, _17;
+                _32.digits = {32};
+                _32.exponent = 1;
+                _48.digits = {48};
+                _48.exponent = 1;
+                _17.digits = {17};
+                _17.exponent = 1;
+
+                reciprocal = _48 - _32*denominator;
+                reciprocal.binary_search_division(_17, maximum_precision); /* approximate division method */
+                /* Initial guess end*/
+
+                --maximum_precision; 
+                exact_number<T> error, max_error, residual, answer, more_precise_answer;
+                max_error.digits = {1};
+                max_error.exponent = (-1)*(maximum_precision);
+                answer = reciprocal * numerator;
+
+                /* newton raphson iteration starts */
+                do{
+                    /* improving guess */
+                    reciprocal = reciprocal * ( _2 - reciprocal*denominator);
+                    reciprocal.normalize();
+
+                    /* truncate insignificant digits from the reciprocal */
+                    while((int) reciprocal.digits.size() - reciprocal.exponent - numerator.exponent >  maximum_precision + 1){
+                        reciprocal.digits.pop_back();
+                    }
 
-                        (*this).normalize();
-                        this->exponent += exponent_dif;
-                        return;
+                    more_precise_answer = reciprocal * numerator;
+                    more_precise_answer.normalize();
+
+                    /* truncate insignificant digits from the reciprocal */
+                    while((int) more_precise_answer.digits.size() - more_precise_answer.exponent > maximum_precision + 1){
+                        more_precise_answer.digits.pop_back();
                     }
-                    // at this point, it is impossible to make the residual 0
-                } // else old residual == 0
 
-                if (positive)
-                    (*this).positive = true;
-                else
-                    (*this).positive = false;
+                    /* if there is no improvement then exit */
+                    if(more_precise_answer == answer){
+                        break;
+                    }
+
+                    /* Compute the improvement in result */
+                    error = more_precise_answer - answer;
+                    error = error.abs();
+
+                    answer = more_precise_answer;
+                    
+                }while( error > max_error );
```

Comment:
Same with style about braces, spaces before or after? same with parenthesis a few lines above you leave no spaces after paretesis but here you do. 
Try to stick to a standard style, if you using an IDE you can probably configure it and make it happen automatically. Otherwise you could configure clang-format and enforce it as part of the commit hook, 
if you do so please share :)

Reply:
I will change that.
I will try to configure the sublime to automate the code formatting.

Reply:
In sublime there are Hinting plugins you can add I think. But As I said before, also be consisten with the library code style, not only with your codes.